gh-68443: Replace debug level-related logic in http client with logging

CuriousLearner opened the pull request:
Hey @bitdancer , here's the PR for issue: https://bugs.python.org/issue24255

<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
bpo-NNNN: Summary of the changes made
```

Where: bpo-NNNN refers to the issue number in the https://bugs.python.org.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `master`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `master`.

-->


<!-- issue-number: [bpo-24255](https://www.bugs.python.org/issue24255) -->
https://bugs.python.org/issue24255
<!-- /issue-number -->


<!-- gh-issue-number: gh-68443 -->
* Issue: gh-68443
<!-- /gh-issue-number -->

dhilst commented on Lib/http/client.py:
```diff
@@ -265,7 +269,7 @@ def _read_status(self):
             raise LineTooLong("status line")
         if self.debuglevel > 0:
-            print("reply:", repr(line))
+            _log.info("reply: {}".format(repr(line)))
```
_logging_ first argument support old format string, so `_log.info("reply: %s", repr(line))` would work. There is mixed calls of `{}".format(foo` with `%s", foo`, wouldn't be better to stick with one?

vsajip replied:
> wouldn't be better to stick with one?

I agree - `%` should work fine and a lot of logging uses %-formatting rather than `.format()` due to its age.

asvetlov replied:
There is another reason: tools like Sentry use message pattern as a key for logs grouping.
Adding pre-formatted messages to Sentry is very ineffective.

CuriousLearner replied:
Ah, sorry!

I think I came back to this later and followed the same style as for other Python projects I was involved with.

Fixing this!

dhilst commented on Lib/test/test_httplib.py:
```diff
@@ -3,6 +3,8 @@
 import io
 import itertools
 import os
+import sys
```
This isn't being used anywhere!?

CuriousLearner replied:
Yeah, good catch. I think it remained here while I was working on this.

vsajip commented on Lib/test/test_logging.py:
```diff
@@ -1954,9 +1954,7 @@ def handle_request(self, request):
-        logger = logging.getLogger("http")
-        root_logger = self.root_logger
-        root_logger.removeHandler(self.root_logger.handlers[0])
+        logger = logging.getLogger("httplogger")
```
Why this name change?

CuriousLearner replied:
There is no specific reason for it. I'll just revert back.

CuriousLearner committed:
bpo-24255: Replace debug prints with logging in http.client

Replace debuglevel-based print statements with Python's standard
logging module for better integration with logging frameworks and
log aggregation tools like Sentry. Uses % formatting consistently
for better log pattern matching in aggregation tools.

CuriousLearner committed:
Merge branch 'main' into fix-issue24255

tpikonen commented on Lib/http/client.py:
```diff
@@ -1062,11 +1078,7 @@ def send(self, data):
                 raise NotConnected()
 
-        if self.debuglevel > 0:
-            print("send:", repr(data))
```
Is this `print` of the data to be sent replaced by something `logging`-based somewhere?

tpikonen commented on Lib/http/client.py:
```diff
@@ -1062,11 +1078,7 @@ def send(self, data):
+        if hasattr(data, "read"):
             encode = self._is_textIO(data)
             if encode and self.debuglevel > 0:
                 print("encoding file using iso-8859-1")
```
I think this PR should replace all `print` calls with `logging`.

tpikonen reviewed (commented):
I ran into the problem (logging with `print`, which gives no timestamps etc.) this PR is fixing, and I hope it gets merged.

But, this patch has several style-related changes, which should be put into a separate commit. It also looks to me that there is some `print` usage left and some debug output which this patch removes, see my comments.
